lib/Listener/NewMessageClassificationListener.php

```diff
@@ -66,11 +80,24 @@ public function handle(Event $event): void {
 			}
 		}
 
-		// if the message is already flagged as important, we won't classify it again.
-		$messages = array_filter($event->getMessages(), function ($message) {
-			return ($message->getFlagImportant() === false);
+		$messages = $event->getMessages();
+
+		// if this is a message that's been flagged / tagged as important before, we don't want to reclassify it again.
+		$doNotReclassify = $this->tagMapper->getTaggedMessageIdsForMessages($event->getMessages(), $event->getAccount()->getUserId(), Tag::LABEL_IMPORTANT);
```

Comment:
```suggestion
		$doNotReclassify = $this->tagMapper->getTaggedMessageIdsForMessages(
				$event->getMessages(),
				$event->getAccount()->getUserId(),
				Tag::LABEL_IMPORTANT
		);
```

makes long lines readable